src/Func.cpp

```diff
@@ -1704,6 +1704,37 @@ Stage &Stage::gpu_tile(VarOrRVar x, VarOrRVar y, VarOrRVar z,
     return gpu_tile(x, y, z, bx, by, bz, tx, ty, tz, x_size, y_size, z_size, tail, device_api);
 }
 
+Stage &Stage::shader(VarOrRVar x, VarOrRVar y, VarOrRVar c, DeviceAPI device_api) {
+    reorder(c, x, y);
+    // GLSL outputs must be stored interleaved
+    // No reorder_storage, so we manually reorder
+    vector<StorageDim> reordered_storage;
+    reordered_storage.push_back(storage_dims[2]);
+    reordered_storage.push_back(storage_dims[0]);
+    reordered_storage.push_back(storage_dims[1]);
+
+    // TODO: Set appropriate constraints if this is the output buffer?
+
+    Stage(definition, stage_name, dim_vars, reordered_storage).gpu_blocks(x, y, device_api);
```

Comment:
This is just following the pattern of the other functions (and is the way it was before); `Stage`'s constructor mutates the definition.  I *have* fixed this to call `Stage::shader` instead of `Stage::gpu_blocks` though.